opencsp/app/sofast/lib/sofast_common_functions.py

```diff
@@ -0,0 +1,232 @@
+    under or over saturated. Displays the crosshairs on the projector once finished.
+
+    Params:
+    -------
+    image_acquisition: ImageAcquisitionAbstract
+        The camera to auto-calibrate the exposure of. If None, then the global instance is used.
+    image_projection: ImageProjection, optional
+        The projector instance to display a flat white image during exposure calibration. If None and a global instance
+        is available, then the global instance is used. Otherwise this step is skipped.
+    on_done: Callable, optional
+        If set, then this callback will be evaluated when the exposure calibration has completed
+
+    Raises:
+    -------
+    RuntimeError:
+        If image_acquisition isn't given and a global instance isn't available.
+    """
+    image_acquisition, image_projection = get_default_or_global_instances(image_acquisition, image_projection)
+
+    # Check that a camera is available
+    check_camera_loaded('run_exposure_calibration', image_acquisition)
+
+    # Try to display a white image on the projector, if there is one
+    if image_projection is None:
+        lt.info('Running calibration without displayed white image.')
+        image_acquisition.calibrate_exposure()
+        if on_done is not None:
+            on_done()
+    else:
+        lt.info('Running calibration with displayed white image.')
+
+        def run_cal():
+            image_acquisition.calibrate_exposure()
+            image_projection.show_crosshairs()
+            if on_done is not None:
+                on_done()
+
+        white_image = np.array(image_projection.zeros()) + image_projection.max_int
+        image_projection.display_image_in_active_area(white_image)
+        image_projection.root.after(100, run_cal)
+
+
+def get_exposure(image_acquisition: ImageAcquisitionAbstract = None) -> int | None:
+    """Returns the exposure time of the camera (microseconds).
+
+    Params
+    ------
+    image_acquisition: ImageAcquisitionAbstract
+        The camera to auto-calibrate the exposure of. If None, then the global instance is used.
+
+    Raises
+    ------
+    RuntimeError:
+        If there is no camera instance
+    """
+    check_camera_loaded('get_exposure', image_acquisition=image_acquisition)
+    image_acquisition, _ = get_default_or_global_instances(image_acquisition_default=image_acquisition)
+    return image_acquisition.exposure_time
+
+
+def set_exposure(new_exp: int, image_acquisition: ImageAcquisitionAbstract = None) -> None:
+    """Sets camera exposure time value to the given value (microseconds)
+
+    Params
+    ------
+    new_exp: int
+        The exposure time to set, in microseconds.
+    image_acquisition: ImageAcquisitionAbstract
+        The camera to set the exposure of. If None, then the global instance is used.
+
+    Raises
+    ------
+    RuntimeError:
+        If there is no camera instance
+    """
+    check_camera_loaded('set_exposure', image_acquisition_default=image_acquisition)
```

Comment:
I believe this should be `image_acquisition`, not `image_acquisition_default`

Reply:
You had me worried, but `image_acquisition_default` is correct.

Reply:
Are you sure (I could definitely be looking at the wrong function)? I'm seeing this inuat as `image_acquisition_default`

https://github.com/bbean23/OpenCSP/blob/ca7a69b915eb6baa522645d0d09f742ecc84d1da/opencsp/app/sofast/lib/sofast_common_functions.py#L73